### PR TITLE
Add custom seed option for speculos backend.

### PR DESCRIPTION
**Pull request by agrojean-ledger**
Add a custom seed option in `configuration.py` to allow the user to use whichever seed he wants with speculos, not only the default one.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Added
+- conftest: add configuration option that allows a custom seed to be used for speculos.
```
Comment (xchapron-ledger) [inline CHANGELOG.md:11]:
```suggestion
- conftest: add configuration and pytest options that allow a custom seed to be used for speculos.
```

File: src/ragger/conftest/base_conftest.py [hunk 2]
```diff
@@ -128,20 +135,28 @@ def prepare_speculos_args(root_pytest_dir: Path, firmware: Firmware, display: bo
                 raise ValueError(f"File '{lib_path}' missing. Did you compile for this target?")
             speculos_args.append(f"-l{lib_name}:{lib_path}")
 
+    # Check if custom user seed has been provided through CLI or optional configuration.
+    # CLI user seed has priority over the optional configuration seed.
+    if cli_user_seed:
+        speculos_args += ["--seed", cli_user_seed]
+    elif not len(conf.OPTIONAL.CUSTOM_SEED) == 0:
```
Comment (xchapron-ledger) [inline src/ragger/conftest/base_conftest.py:142]:
```suggestion
    elif conf.OPTIONAL.CUSTOM_SEED:
```
Should work no?

File: CHANGELOG.md (hunk 1 above)
Comment (agrojean-ledger) [reply CHANGELOG.md:11]: Ok

**Comment by xchapron-ledger** [comment]
A small rebase and we should be good!

File: src/ragger/conftest/base_conftest.py (hunk 2 above)
Comment (agrojean-ledger) [reply src/ragger/conftest/base_conftest.py:142]: @xchapron-ledger I rely on the same syntax that @fbeutin-ledger used for other `conf.OPTIONAL` usage. I think I remember he had some good point to use this one...

**Review by xchapron-ledger — APPROVED** [review]

**Pull request merged**